app/servers/query_master_tcp.py

```
@@ -0,0 +1,378 @@
+        logger.debug("Processing packet: %d bytes", len(packet))
+
+        try:
+            # Get the handler and process the query
+            handler = self.get_handler()
+
+            # Update handler's master IP based on actual server IP if possible
+            if self.peername:
+                sock = self.transport.get_extra_info("socket")
+                if sock:
+                    local_addr = sock.getsockname()
+                    if local_addr[0] != "0.0.0.0":
```

Ruff (0.14.10)

Possible binding to all interfaces

(S104)